### PR TITLE
 Add a setup step to clone/pull the repo

### PR DESCRIPTION
**Pull request by thomasleese**
This takes us towards the idea of being able to do `gem install govuk-docker` because the next step can be to do `govuk-docker install` and the repo will be cloned automatically.

### CODE REVIEW

**Comment by benthorner** [comment]
I wonder if 'setup' might work better here? I'm struggling with seeing the two 'install' commands together and have to stop and think about what the difference is. Might help to clarify here what we expect the gem to do vs. the cloned repo.

**Comment by thomasleese** [comment]
@benthorner Are you thinking we have two separate commands, `setup` and `install`? Or that we change the name of the command to `setup`?

The idea behind `install` is that you can run it at any point and it will install anything missing to successfully run govuk-docker. So in most cases, it won't reclone the repo since it'll already exist. We're planning to add another `install` step shortly which will install Docker itself for you.

**Comment by benthorner** [comment]
@thomasleese yes to the first question. I think it depends on the big picture. Do we have a complete list of commands we're aiming for? It's actually the 'gem install' that confuses me, as I'm struggling to see the benefit of doing that over 'git clone ...'. Playing a devil's advocate somewhat: "why do I need to install it after I've just installed it" is the question that comes to mind ;-).

**Comment by thomasleese** [comment]
I guess one of the benefits is that you don't need to change your `PATH` to get it working, and it'll work under many different shells (right now the instructions only list how to do it for bash). Most respondents to the survey requested this method of installation so perhaps it's worth asking why.

I don't like the idea of two different installation-like commands because I think it'll get confusing, you'll have to ask yourself "which install command do I run to get the latest Dockerfiles? which one do I run if dnsmasq isn't working?".

At the moment the complete list is: Dnsmasq, Docker and the repo. Enough for the entire installation to be `gem install govuk-docker && govuk-docker install`.

I'm not opposed to renaming it to `govuk-docker setup` though, and I think that does make more sense if you look at it together with `gem install govuk-docker`.

**Comment by benthorner** [comment]
Ah, the use of PATH would indeed be an advantage. It looks like we're asking similar questions - perhaps it's worth listing all the command we have / aim to have, so we can understand what they look like together and where potential confusion might occur, just like the scenario you mentioned.

For example, I think the next command after 'install'/'setup' would be something like 'build', possibly in a repo or with an arg/flag to specify the service; but would this too look confusing after 'install'/'setup'?

File: lib/setup/repo.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+require_relative "./base"
+require_relative "../doctor/dnsmasq"
+
+class Setup::Repo < Setup::Base
+  def initialize(*args)
+    super(*args)
+    @path = File.join(ENV.fetch("HOME"), "govuk", "govuk-docker")
```
Comment (barrucadu) [inline lib/setup/repo.rb]: We've got environment variable inconsistency between `govuk-docker` and the Makefile: https://github.com/alphagov/govuk-docker/blob/f56b7cc51f0ee9223b8ae2d6f158ece2774005de/Makefile#L1-L3

Can we have the same environment variables, with the same defaults?  The advantage of not hard-coding `File.join(ENV.fetch("HOME"), "govuk", "govuk-docker")` is that you can do things in a different directory.

Comment (thomasleese) [reply lib/setup/repo.rb]: Yep, good call... I've added that in now.

**Pull request merged**